### DISCUSSION TITLE
keywords and description seems to come from the last published version not actual version.

### DISCUSSION THREAD

**Discussion by eadf** [rust-lang/crates.io#7560] [Feature Requests]
### Current Behavior

I have a crate published at version 0.12.0.
I recently released a bugfix/version bump 0.7.1 with an older description.
Now my 0.12.0 release has description and keywords from the 0.7.1 version.

### Expected Behavior

I expect the description and keywords of one published crate version to be as published.

### Steps To Reproduce

Release/publish a crate with one set of keywords and description. 
Amend an older version of the same crate with other keywords and description.
Expect the newest (highest version) crate to keep its keywords and description as published.

### Environment

- Browser: Safari
- OS: macOs

### Anything else?

~affected crate: https://crates.io/crates/linestring~ (had to push 0.12.1 resetting the problem)

**Comment by 0xPoe** [comment]
I can reproduce this issue. I am working on this.

**Comment by 0xPoe** [comment]
I am not sure if this is by design or a flaw in our current design. We only have a connection between crates and keywords instead of versions.

| crate\_id | keyword\_id |
| :--- | :--- |
| 1 | 1 |
| 1 | 6 |

Every time you want to publish a new version, it will update it to use the latest keywords.
https://github.com/rust-lang/crates.io/blob/04ae0696053aa85abe0ff0720d66c7e03d120e42/src/models/keyword.rs#L74

**Reply by Turbo87** [reply]
> I am not sure if this is by design or a flaw in our current design.

This is currently working as designed. Most of the per-crate metadata is updated whenever a new version is released, regardless of the version number.

The only potential improvement I could see is that the metadata should come from the "default" version of the crate (the highest non-pre-release version). Since this is a user-facing change it might need an RFC though.

**Comment by 0xPoe** [comment]
It seems we should build a connection between versions and keywords.

**Comment by tyilo** [comment]
This is especially a problem for my crate, [fibonacci-numbers](https://crates.io/crates/fibonacci-numbers), where every version has a different description.
I would expect https://crates.io/crates/fibonacci-numbers/185.0.0 to say "The 185th Fibonacci number" and not "The 186th Fibonacci number".

**Reply by Turbo87** [reply]
> We do not allow content or activity on crates.io that:
> - is abusing the package index for purposes it was not intended

(https://crates.io/policies) 😉
